src/Summoner/Template/GitHub.hs

```diff
@@ -118,23 +99,35 @@ gitHubFiles Settings{..} =
 
         matrix:
           include:
-
           $travisCabalMtr
           $travisStackMtr
 
         $installAndScript
 
         notifications:
           email: false
-        $endLine
         |]
 
+    travisCabalCache, travisStackCache :: Text
+    travisCabalCache = memptyIfFalse settingsCabal "- \"$HOME/.cabal\""
+    travisStackCache = memptyIfFalse settingsStack
+        [text|
+        - "$$HOME/.stack"
+        - "$$TRAVIS_BUILD_DIR/.stack-work"
+        |]
+
+
     cabalTest :: Text
     cabalTest = if settingsTest then "cabal new-test" else "echo 'No tests'"
 
+    travisCabalMtr :: Text
+    travisCabalMtr = memptyIfFalse settingsCabal $
+        T.concat $ map travisCabalMatrixItem settingsTestedVersions
```

Comment:
What do you think about having the following function:

```haskell
tconcatMap :: (a -> Text) -> [a] -> Text
tconcatMap f = T.concat . map f
```

Not necessary in this PR (I can create simple and small refactoring issue for that) but I see that we use such pattern a lot, so probably can add this in `Summoner.Text` module and replace all occurrences of this.